plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/jdbc/IcebergJdbcClient.java

```diff
@@ -25,14 +26,25 @@ public class IcebergJdbcClient
 {
     private final IcebergJdbcConnectionFactory connectionFactory;
     private final String catalogName;
+    private final SchemaVersion schemaVersion;
 
-    public IcebergJdbcClient(IcebergJdbcConnectionFactory connectionFactory, String catalogName)
+    public IcebergJdbcClient(IcebergJdbcConnectionFactory connectionFactory, String catalogName, SchemaVersion schemaVersion)
     {
         this.connectionFactory = requireNonNull(connectionFactory, "connectionFactory is null");
         this.catalogName = requireNonNull(catalogName, "catalogName is null");
+        this.schemaVersion = requireNonNull(schemaVersion, "schemaVersion is null");
     }
 
     public void createTable(String schemaName, String tableName, String metadataLocation)
+    {
+        switch (schemaVersion) {
+            case SchemaVersion.V0 -> createTableV0(schemaName, tableName, metadataLocation);
+            case SchemaVersion.V1 -> createTableV1(schemaName, tableName, metadataLocation);
+            default -> throw new IllegalArgumentException("Unsupported schema version: " + schemaVersion);
```

Comment:
Please remove this default clause. https://github.com/trinodb/trino/blob/master/.github/DEVELOPMENT.md#avoid-default-clause-in-exhaustive-enum-based-switch-statements

Reply:
Done!